### ISSUE TITLE
editor bug in algorithms chapter?

### ISSUE THREAD

**Issue by longouyang** [probmods/ppaml2016#1]
If you first run the first example, which is intentionally errorful:

``` js
var gaussianModel = function() {
    return gaussian(0, 1)
};
Infer({method: 'enumerate'}, gaussianModel);
```

and then run this second one, which should work just fine:

``` js
///fold:
var drawLines = function(canvas, start, positions){
  if (positions.length == 0) { return []; }
  var next = positions[0];
  canvas.line(start[0], start[1], next[0], next[1], 4, 0.2);
  drawLines(canvas, next, positions.slice(1));
};

var drawPoints = function(canvas, positions){
  if (positions.length == 0) { return []; }
  var next = positions[0];
  canvas.circle(next[0], next[1], 2, "red", "white");
  drawPoints(canvas, positions.slice(1));
};
///

var genObservation = function(pos){
  return map(
    function(x){ return gaussian(x, 5); },
    pos
  );
};

var init = function(){
    var state1 = [gaussian(200, 1), gaussian(200, 1)];
    var state2 = [gaussian(200, 1), gaussian(200, 1)];
    var states = [state1, state2];
    var observations = map(genObservation, states);
    return {
        states: states,
        observations: observations
    };
};

var transition = function(lastPos, secondLastPos){
  return map2(
    function(lastX, secondLastX){
      var momentum = (lastX - secondLastX) * .7;
      return gaussian(lastX + momentum, 3);
    },
    lastPos,
    secondLastPos
  );
};

var trajectory = function(n) {
  var prevData = (n == 2) ? init() : trajectory(n - 1);
  var prevStates = prevData.states;
  var prevObservations = prevData.observations;
  var newState = transition(last(prevStates), secondLast(prevStates));
  var newObservation = genObservation(newState);
  return {
    states: prevStates.concat([newState]),
    observations: prevObservations.concat([newObservation])
  }
};

var numSteps = 80;
var synthObservations = trajectory(numSteps).observations;
var canvas = Draw(400, 400, true)
drawPoints(canvas, synthObservations)
```

You actually get the error from the first example with a weird error highlighting.

I think there must be a bug somewhere in the error-handling + trampolining code.

**Comment by null-a** [comment]
I think the problem is that after the first error, `env.coroutine`  is left as an instance of `Enumerate`. I'll work on a fix.

**Comment by null-a** [comment]
Closed by probmods/webppl#565.
